src/common/prntbase.cpp

```diff
@@ -949,6 +949,9 @@ wxPreviewCanvas::~wxPreviewCanvas()
 
 void wxPreviewCanvas::OnPaint(wxPaintEvent& WXUNUSED(event))
 {
+    if ( m_printPreview )
+        m_printPreview->UpdatePageRendering();
+
```

Comment:
I'm really not sure this should be done here. There is a comment saying that `UpdatePageRendering()` is called during idle time only and while I don't fully understand the logic of doing this, it seems wrong to call it both from there and from here.

If we need to update the scale factor, we should just call some (private) `DoUpdateScaleFactorIfNeeded()` but OTOH I don't even know why should we do this from here and not during initialization and then from `wxEVT_DPI_CHANGED` handler?

Reply:
> I don't even know why should we do this from here and not during initialization and then from `wxEVT_DPI_CHANGED` handler?

Yes, it should be better to call it from `wxEVT_DPI_CHANGED` handler. I'm just not sure what's the proper way to not break the ABI here.

Reply:
Adding a new (non-virtual, private) function doesn't break ABI, so we could add `wxEVT_DPI_CHANGED` handler and `Bind()` to it.